bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,872 @@
+                    getQuantityTypeState(data.getNightProbabilityOfPrecipitation(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_SNOW,
+                    getQuantityTypeState(data.getDayProbabilityOfSnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_SNOW,
+                    getQuantityTypeState(data.getNightProbabilityOfSnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_HEAVY_SNOW,
+                    getQuantityTypeState(data.getDayProbabilityOfHeavySnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_HEAVY_SNOW,
+                    getQuantityTypeState(data.getNightProbabilityOfHeavySnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_RAIN,
+                    getQuantityTypeState(data.getDayProbabilityOfRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_RAIN,
+                    getQuantityTypeState(data.getNightProbabilityOfRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_HEAVY_RAIN,
+                    getQuantityTypeState(data.getDayProbabilityOfHeavyRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_HEAVY_RAIN,
+                    getQuantityTypeState(data.getNightProbabilityOfHeavyRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_HAIL,
+                    getQuantityTypeState(data.getDayProbabilityOfHail(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_HAIL,
+                    getQuantityTypeState(data.getNightProbabilityOfHail(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_SFERICS,
+                    getQuantityTypeState(data.getDayProbabilityOfSferics(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_SFERICS,
+                    getQuantityTypeState(data.getNightProbabilityOfSferics(), Units.PERCENT));
+        }
+    }
+
+    private void handleNon200Response(final Response resp) {
+        // Handle failed credentials
+        if (resp.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+            // Remove this once the status is updated accordingly
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    getLocalizedText("bridge.error.site-specific.auth-issue"));
+            authFailed = true;
+        }
+    }
+
+    private @Nullable MetOfficeDataHubBridgeHandler getMetOfficeDataHubBridge() {
+        Bridge baseBridge = getBridge();
+
+        if (baseBridge != null && baseBridge.getHandler() instanceof MetOfficeDataHubBridgeHandler) {
+            return (MetOfficeDataHubBridgeHandler) baseBridge.getHandler();
```

Comment:
```suggestion
        if (baseBridge != null && baseBridge.getHandler() instanceof MetOfficeDataHubBridgeHandler bridgeHandler) {
            return bridgeHandler;
```

Reply:
Thank you I thought that I had found all of these. 👍